### ISSUE TITLE
Question on masking in obgn-arxiv/gcn

### ISSUE THREAD

**Issue by mearcstapa-gqz** [dmlc/dgl#2904]
Hi! I'm wondering why the masking is being used? I think the effect is that it chooses different fraction of nodes for loss calculation in training. So why not use all nodes for loss calculation? Is this for improving generalizing ability? And is it a widely used practice? Can you refer me to some papers on it?
Thanks a lot!!
https://github.com/dmlc/dgl/blob/103444c58f66bd7a25407cb8fba6248d8cc976e5/examples/pytorch/ogb/ogbn-arxiv/gcn.py#L68-L72

**Comment by BarclayII** [comment]
@Espylapiza Maybe you can give more insights?

**Comment by Espylapiza** [comment]
@mearcstapa-gqz
Thanks for your question. This means only a fraction of the data samples are selected for updating and it behaves similarly to stochastic gradient descent. When using stochastic gradient descent many people have discussed the choice of batch size, and large batch sizes are often not a good choice, let alone full-batch training. Some papers also investigated why stochastic gradient descent is more robust to generalization than gradient descent, e.g. https://arxiv.org/abs/1609.04836. Hope this is what you are looking for.

However, I am not sure if it leads to better results here since I have not done ablation studies for this.

**Comment by skepsun** [comment]
Besides of behaving similarly to stochastic gradient descent, it is also a part of "label masking in train set" (use-label trick).
